src/server/game/Groups/Group.cpp

```diff
@@ -2092,7 +2092,39 @@ void Group::ResetInstances(uint8 method, bool isRaid, Player* SendMsgTo)
         {
             // do not reset the instance, just unbind if others are permanently bound to it
             if (isEmpty && instanceSave->CanReset())
+            {
+                if (map && this->isRaidGroup() && map->IsDungeon())
+                {
+                    AreaTrigger const * const instanceEntrance = sObjectMgr->GetGoBackTrigger(map->GetId());
+
+                    if (!instanceEntrance)
+                        TC_LOG_DEBUG("root", "Instance entrance not found for maps %u", map->GetId());
+                    else
+                    {
+                        const uint32 zoneId = sMapMgr->GetZoneId(instanceEntrance->target_mapId, instanceEntrance->target_X, instanceEntrance->target_Y, instanceEntrance->target_Z);
+
+                        for (const MemberSlot &member : this->GetMemberSlots())
+                        {
+                            if (!ObjectAccessor::FindPlayer(member.guid))
```

Comment:
I think you might have contradiction here.

If one of the players in entering some other map (teleporting to it) `ObjectAccessor::FindPlayer` will not find him therefore it will teleport him back to the instance entrance. I think you should better use `ObjectAccessor::FindConnectedPlayer` instead.

Reply:
Not to mention this will also teleport all players from anywhere in the world, not just if they were in the instance already

Reply:
@kuskmen Yes, thank you, I missed it

@Shauren That is what I said (or at least tried to said) in the PR description.
I must find a way to determine whithin the offline players list which of them are in the instance.
I would like to have your suggestions about it.

Reply:
I think you can get members guid and mapId and check in `characters.characters` if mapId of the instance is the same as the one in the `characters.characters` table?

Reply:
Create a new query with additional `map_id != ?` restriction in WHERE part

Reply:
Ok, I'll do that. And add it to CharacterDatabase

Reply:
iirc people inside instances where send to graveyards on wotlk not to instance entrance.